company.el

```diff
@@ -2139,6 +2139,14 @@ each one wraps a part of the input string."
       (cl-incf i))))
 
 (defun company-search-keypad ()
+  "Prevent disabling of `company-search-mode' on a keypad number press.
```

Comment:
The first line of a docstring should be a (short) complete sentence which describes what the command does.

The rest can explain further details, including, like in this example, the reasons for the command's existence.

As far as I'm concerned, this explanation can go in a comment somewhere, for example inside the `company-search-map`'s definition above the line where `'company-search-keypad` is bound.

But if you want to add it in the docstring, let's make the first line something like:

> Handle keypress on the keypad and append the typed key to the search string.

`company-search-printing-char` could probably use a docstring as well.